app/src/main/java/com/example/una/SearchFragment.java

```diff
@@ -0,0 +1,107 @@
+import org.json.JSONException;
+import java.util.ArrayList;
+
+import cz.msebera.android.httpclient.Header;
+
+public class SearchFragment extends Fragment {
+
+    // the base URL for the API
+    public final static String API_BASE_URL = "https://api.data.charitynavigator.org/v2";
+    // the parameter name for the API key
+    public final static String API_KEY_PARAM = "app_key";
+    // the parameter name for the API application ID
+    public final static String API_ID_PARAM = "app_id";
+    public final static String TAG = "SearchFragment";
+
+
+    protected RecyclerView rvCategories;
+    protected CategoryAdapter adapter;
+    protected ArrayList<Category> categories;
+    AsyncHttpClient client;
+
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_search, container, false);
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        rvCategories = view.findViewById(R.id.rvCategories);
+        client = new AsyncHttpClient();
+
+        // create the data source
+        categories = new ArrayList<>();
+
+        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(view.getContext());
+        adapter = new CategoryAdapter(categories);
+        rvCategories.setAdapter(adapter);
+        rvCategories.setLayoutManager(linearLayoutManager);
+
+        getCategories();
+    }
+
+    private void getCategories() {
+        // create the url
+        String url = API_BASE_URL + "/Categories";
+
+        RequestParams params = new RequestParams();
+        params.put(API_KEY_PARAM, getString(R.string.cn_api_key));
+        params.put(API_ID_PARAM, getString(R.string.cn_app_id));
+
+        client.get(url, params, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int statusCode, Header[] headers, JSONArray response) {
+                try {
+                    for (int i = 0; i < response.length(); i++) {
+                        Category category = new Category(response.getJSONObject(i));
+                        categories.add(category);
+                        adapter.notifyItemInserted(categories.size()-1);
+                        Log.i(TAG, String.format("Loaded %s categories", categories.size()));
+
+                    }
+                } catch (JSONException e) {
+                    logError("Failed to parse categories", e, true);
+                }
+            }
+
+            @Override
+            public void onFailure(int statusCode, Header[] headers, String responseString, Throwable throwable) {
+                logError("Failed to get data from categories endpoint", throwable, true);
+            }
+        });
+    }
+
+    // handle errors, log and alert user
+    private void logError(String message, Throwable error, boolean alertUser) {
```

Comment:
is `alertUser` ever false?

Reply:
No, I don't think so, I'll delete the parameter.